### PR TITLE
Native: Add azure ad page

### PR DESCRIPTION
**Pull request by dotNetkow**

### CODE REVIEW

File: src/pages/enterprise/auth-connect/azure-ad.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+```
+
+Some of these `IonicAuthOptions` values are unique, and must be set based on your Azure AD app’s details:
+
+* `platform`: Use “cordova” or “capacitor” accordingly.
+* `clientID`: Your app’s _Application ID_. Example: cebbb0be-d578-4bbd-9712-4b0fe05c06aa
+* `redirectUri`: The URI to redirect to after the user has logged in. Use the same AUTH_URL_SCHEME variable value (App Id) from when the Auth Connect plugin was installed. Example: com.company.app://callback
+* `logoutUrl`: The URI to redirect to after the user has logged out. Example: com.company.app://login?logout=true
+* `audience`: Your custom API URL of choice, such as `https://api.myapp.com`.
+* `scope`: Unlock access to protected resources, such as read/write permissions. Example: openid offline_access email picture profile
```
Comment (dotNetkow) [inline src/pages/enterprise/auth-connect/azure-ad.md]: Scope: in the CS Demo app, we also have "https://vsquad.onmicrosoft.com/api/Hello.Read", which appears to be a custom Published Scope in our private Azure org. Is something like this required, or is the default assigned "user_impersonation" good enough?

File: src/pages/enterprise/auth-connect/azure-ad.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+The `discoveryUrl` consists of the following format:
+
+`https://B2C-TENANT-NAME.b2clogin.com/B2C-TENANT-NAME.onmicrosoft.com/v2.0/.well-known/openid-configuration?p=POLICY-NAME`
+
+Replace `B2C-TENANT-NAME` with your tenant name and the `POLICY-NAME` with the name of the User Flow created earlier.
```
Comment (dotNetkow) [inline src/pages/enterprise/auth-connect/azure-ad.md]: Did I explain this correctly? Got the explanation from [here](https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-add-identity-providers#create-an-azure-active-directory-application).

File: src/pages/enterprise/auth-connect/azure-ad.md (hunk 1 above)
Comment (lm3m) [reply src/pages/enterprise/auth-connect/azure-ad.md]: you need at least `offline_access` that one is required to use tokens.

File: src/pages/enterprise/auth-connect/azure-ad.md (hunk 2 above)
Comment (lm3m) [reply src/pages/enterprise/auth-connect/azure-ad.md]: the other place to find it is on the azure portal:
azure ad b2c-> app -> user flows (policies) -> run user flow -> the discovery link is at the top

also for non-b2c is is a bit easier.

File: src/components/menu/templates/enterprise.tsx [hunk 3]
```diff
@@ -12,7 +12,8 @@ const items = {
   'menu-enterprise-authentication-security': {
     'Auth Connect': '/docs/enterprise/auth-connect',
     '': {
-      'Auth0': '/docs/enterprise/auth-connect/auth0'
+      'Auth0': '/docs/enterprise/auth-connect/auth0',
+      'Azure AD': '/docs/enterprise/auth-connect/azure-ad'
```
Comment (lm3m) [inline src/components/menu/templates/enterprise.tsx]: we might want to call this Azure AD B2C, not v2.

File: src/pages/enterprise/auth-connect/azure-ad.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+With that in hand, set the _Custom Redirect URI_. After the app user signs into Azure AD, this tells Auth Connect which page to redirect to in your app. Use the formula “uniqueId://page”, such as `com.company.app://callback`.
+
+After filling in all details above, click the Create button. 
+
+#### Create User Flows (Policies)
+
+Create at least one [User Flow](https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-create-user-flows), the series of pages your users will interact with to sign up/sign into their account. Note the `Name` of the User Flow you create, as it's needed for configuring the Auth Connect `Discovery URL` property.
+
+Azure AD is now ready to use with Auth Connect.
+
```
Comment (lm3m) [inline src/pages/enterprise/auth-connect/azure-ad.md]: might comment on noting the discovery url while creating the flow(s).
also, the discovery url for both password reset and create/login should both be noted.

**Comment by lm3m** [comment]
password reset is one other area we should note, I will send you an email on that one. it has some interesting complexity.

File: src/components/menu/templates/enterprise.tsx (hunk 3 above)
Comment (dotNetkow) [reply src/components/menu/templates/enterprise.tsx]: ah good call. done.

File: src/pages/enterprise/auth-connect/azure-ad.md (hunk 4 above)
Comment (dotNetkow) [reply src/pages/enterprise/auth-connect/azure-ad.md]: good call, added. thanks!

File: src/pages/enterprise/auth-connect/azure-ad-b2c.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+Where `B2C-TENANT-NAME` is your tenant name and the `POLICY-NAME` is the name of the User Flow created earlier.
+
+### What's Next?
+
+Check out the full list of [configuration options](/docs/enterprise/auth-connect#ionicauthoptions) available, then implement the [other steps](/docs/enterprise/auth-connect#workflow) in the Auth Connect workflow.
+
+## Implementing password reset using custom user flows/policies
+
+To implement password reset functionality, a custom User Flow needs to be created. Navigate to the `User flows (policies)` page, then click the "New user flow" button. Next, select the "Password reset" user flow type. As part of the `Application claims` section, choose "Email Addresses" at a minimum. After the user flow has been created, select it from the User Flow list, then click "Run user flow" from the Overview tab. Note the URL at the top of the page - use it as the discovery url for password reset.
```
Comment (dotNetkow) [inline src/pages/enterprise/auth-connect/azure-ad-b2c.md]: Moved this from the main Auth Connect page to here, and expanded on the details. including, adding a code snippet.

**Pull request merged**